samza-api/src/main/java/org/apache/samza/operators/StreamGraphBuilder.java

```diff
@@ -27,6 +27,8 @@
  */
 @InterfaceStability.Unstable
 public interface StreamGraphBuilder {
+  static final String BUILDER_CLASS_CONFIG = "job.graph.builder.class";
```

Comment:
Good points. I am not sure what's the new scope that we are creating for pipeline. I can probably use that. @jmakes and Xinyu, any suggestions?